### PR TITLE
Speedup of PrimePerl/solution_1. Same algorithm.

### PR DESCRIPTION
**Pull request by kjetillll**
## Contributing requirements

* [x] I read the contribution guidelines in CONTRIBUTING.md.
* [x] I placed my solution in the correct solution folder.
* [x] I added a README.md with the right badge(s).
* [x] I added a Dockerfile that builds and runs my solution.
* [x] I selected `drag-race` as the target branch.
* [x] All code herein is licensed compatible with BSD-3.

### CODE REVIEW

File: PrimePerl/solution_2/primes.pl [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                substr $$bits, $num2, 1, '1';
+                $num2 += $factor2;
+            }
+            $factor += 2;
+        }
+    }
+
+    sub primes {
+        my $self = shift;
+	my $bits=\ $self->{bits};
```
Comment (marghidanu) [inline PrimePerl/solution_2/primes.pl]: Why the double reference? I see you're dereferencing it immediately in the next function.

File: PrimePerl/solution_2/primes.pl [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        100000000   => 5761455,
+        1000000000  => 50847534,
+        10000000000 => 455052511,
+    );
+
+    sub new {
+        my ( $class, $sieve_size ) = @_;
+	bless {
+            sieve_size => $sieve_size,
+	    bits       => '0' x $sieve_size,
```
Comment (marghidanu) [inline PrimePerl/solution_2/primes.pl]: Strings? :) That's interesting...

File: PrimePerl/solution_2/primes.pl (hunk 1 above)
Comment (kjetillll) [reply PrimePerl/solution_2/primes.pl]: So not to copy it. It's a long string, a million chars in the default case. Saves doing lots of hash-lookups on 'bits'. Even if it's just a two keys hash, hash-lookups can be slower than dereferencing a string-ref I think.

File: PrimePerl/solution_2/primes.pl (hunk 2 above)
Comment (kjetillll) [reply PrimePerl/solution_2/primes.pl]: Perl is good at strings, got a speedup from it.

**Comment by bbrtj** [comment]
From ~50 to ~440 passes on my machine. Good job!

**Comment by rbergen** [comment]
@marghidanu What's your opinion on this PR, as it stands?

**Comment by marghidanu** [comment]
It's ok, it's basically an optimization for dereferencing variables.

**Comment by rbergen** [comment]
Do you agree this should be added as a second solution, or is this an optimization that should be applied to the existing solution_1?

**Comment by marghidanu** [comment]
It's ok in solution_1, nothing else has changed.

**Comment by bbrtj** [comment]
@marghidanu are you sure you're looking at the right PR? This one is about using string as a storage for sieve data, not about dereferencing. I think you might've confused it with the other PR that is now merged.

**Comment by marghidanu** [comment]
My bad, I was talking about the other one :). Sorry about that, this one is ok too. Actually, I'm quite surprised this is so fast just by using a string as storage.

**Comment by rbergen** [comment]
So this should be applied to solution_1 then?

**Comment by marghidanu** [comment]
No, let's keep it as solution_2

**Pull request merged**